src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js

```diff
@@ -216,36 +245,48 @@ class ActivityFeed extends React.Component<Props, State> {
                             <LoadingIndicator />
                         </div>
                     )}
+                    {!isLoading && (
+                        <ul className="bcs-activity-feed-active-state">
+                            {isEmpty && <EmptyState showCommentMessage={showCommentForm} />}
+                            {!isEmpty &&
+                                collapseFeedState(feedItems).map((item: FeedItem) => {
+                                    const { type, id } = item;
+                                    const isFocused = item === activeEntry;
+                                    const refValue = isFocused ? this.activeFeedItemRef : { current: null };
 
-                    {isEmpty && !isLoading && <EmptyState showCommentMessage={showCommentForm} />}
-
-                    {!isEmpty && !isLoading && (
-                        <ActiveState
-                            {...activityFeedError}
-                            items={collapseFeedState(feedItems)}
-                            isDisabled={isDisabled}
-                            currentUser={currentUser}
-                            onTaskAssignmentUpdate={onTaskAssignmentUpdate}
-                            onAppActivityDelete={onAppActivityDelete}
-                            onCommentDelete={hasCommentPermission ? onCommentDelete : noop}
-                            onCommentEdit={hasCommentPermission ? onCommentUpdate : noop}
-                            // We don't know task edit/delete specific permissions,
-                            // but you must at least be able to comment to do these operations.
-                            onTaskDelete={hasCommentPermission ? onTaskDelete : noop}
-                            onTaskEdit={hasCommentPermission ? onTaskUpdate : noop}
-                            onTaskModalClose={onTaskModalClose}
-                            onVersionInfo={onVersionHistoryClick ? this.openVersionHistoryPopup : null}
-                            translations={translations}
-                            getAvatarUrl={getAvatarUrl}
-                            getUserProfileUrl={getUserProfileUrl}
-                            mentionSelectorContacts={mentionSelectorContacts}
-                            getMentionWithQuery={getMentionWithQuery}
-                            approverSelectorContacts={approverSelectorContacts}
-                            getApproverWithQuery={getApproverWithQuery}
-                            activeFeedEntryId={activeFeedEntryId}
-                            activeFeedEntryType={activeFeedEntryType}
-                            activeFeedItemRef={this.activeFeedItemRef}
-                        />
+                                    return (
+                                        <ActivityFeedItem
```

Comment:
I'm not sure that I love how `ActivityFeedItem` has to take in all the possible props for the various feed item types, somehow it felt more reasonable for `ActiveState`

Reply:
I agree, I don't think this is the direction we want to go. If we want to refactor here, it seems like each type of activity item should be moved to its own component first. We should avoid creating more mega-components.